### PR TITLE
Add/54020 global unique id orders endpoint

### LINKED ISSUE #54020: [Enhancement]: Add global_unique_id to /orders and /order/{id} Rest API endpoints
### Describe the solution you'd like

We really need the global_unique_id field in the /orders and /order/{id} endpoint. This way, we can identify which product has been ordered within our system. We need this so we can support the global_unique_id native, instead of the custom EAN packages.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jobjen02**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes #54020.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Go with the Rest API to the /orders uri.
2. See that the global_unique_id appeared in the `line_items` list.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [x] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

**Comment by jobjen02** [comment]
I also did a small cleanup there, as you can see. Before there was a double check to see if the product exists. It's now in one check with the product variable added, to increase performance.

**Comment by jobjen02** [comment]
@naman03malhotra, when are you albe to check this? This is an really important fix for our warehousing software. Thanks!

**Comment by jobjen02** [comment]
@naman03malhotra, when are you albe to check this? This is an really important fix for our warehousing software. Thanks!

File: plugins/woocommerce/includes/rest-api/Controllers/Version2/class-wc-rest-orders-v2-controller.php [hunk 1]
```diff
@@ -223,15 +223,27 @@ protected function get_order_item_data( $item ) {
...
+				$data['image'] = array(
+					'id'  => $image_id,
+					'src' => $image_id ? wp_get_attachment_image_url( $image_id, 'full' ) : '',
+				);
+
+				if ( is_callable( array( $product, 'get_parent_data' ) ) ) {
+					$data['parent_name'] = $product->get_title();
+				} else {
+					$data['parent_name'] = null;
+				}
```
Comment (naman03malhotra) [inline plugins/woocommerce/includes/rest-api/Controllers/Version2/class-wc-rest-orders-v2-controller.php]: I am yet to review this PR in depth, however I am curious why include this change as part of this PR?

Comment (jobjen02) [reply plugins/woocommerce/includes/rest-api/Controllers/Version2/class-wc-rest-orders-v2-controller.php]: Thanks! I updated that because there was some duplicate code for that part. I just moved some lines, but git displays it as an big change.

I can revert that part if you prefer that?

Comment (jobjen02) [reply plugins/woocommerce/includes/rest-api/Controllers/Version2/class-wc-rest-orders-v2-controller.php]: @naman03malhotra I just restored the original code, so only my change is there now :)

**Review by naman03malhotra — COMMENTED** [review]
Thanks for making the changes. I am also curious about your usecase of needing global id with orders endpoint.

**Comment by jobjen02** [comment]
> Thanks for making the changes. I am also curious about your usecase of needing global id with orders endpoint.

We develop warehouse software for customers. Every product with us has an ean code on which we match everything. For example, as soon as a Shopify order comes in, we get the ean and we know which products have been ordered. With WooCommerce, this was also possible with an EAN plugin, then this ean was put in the meta_data by the plugin. But now that WooCommerce itself supports EAN codes, this is no longer in the meta data, but we need it natively.

In short, based on the EAN code we know which product it is.

**Comment by jobjen02** [comment]
> Thanks for making the changes. I am also curious about your usecase of needing global id with orders endpoint.

@naman03malhotra 

We develop warehouse software for customers. Every product with us has an ean code on which we match everything. For example, as soon as a Shopify order comes in, we get the ean and we know which products have been ordered. With WooCommerce, this was also possible with an EAN plugin, then this ean was put in the meta_data by the plugin. But now that WooCommerce itself supports EAN codes, this is no longer in the meta data, but we need it natively.

In short, based on the EAN code we know which product it is.

**Review by naman03malhotra — DISMISSED** [review]
Thanks for explaining. Can you please
- Add this change to the [schema](https://github.com/naman03malhotra/woocommerce/blob/trunk/plugins/woocommerce/includes/rest-api/Controllers/Version2/class-wc-rest-orders-v2-controller.php#L1142) as well.
- Please update the tests accordingly.

**Comment by hafizubikm** [comment]
This is also really important in terms of being able to look up/search for products by GTIN for inventory control etc.

**Comment by naman03malhotra** [comment]
I've reopned the PR and assigned Rubik to review as the original issues is assigned to their team.

File: plugins/woocommerce/includes/rest-api/Controllers/Version2/class-wc-rest-orders-v2-controller.php [hunk 2]
```diff
@@ -1616,6 +1617,12 @@ public function get_item_schema() {
 								'context'     => array( 'view', 'edit' ),
 								'readonly'    => true,
 							),
+							'global_unique_id' => array(
+								'description' => __( 'Product global unique id.', 'woocommerce' ),
```
Comment (mikejolley) [inline plugins/woocommerce/includes/rest-api/Controllers/Version2/class-wc-rest-orders-v2-controller.php]:
```suggestion
								'description' => __( 'GTIN, UPC, EAN or ISBN.', 'woocommerce' ),
```

**Review by mikejolley — CHANGES_REQUESTED** [review]
This looks good actually and since its present in products routes its fine to include.

I do however think we need to match the product route in the naming of the field in schema. Ive submitted a suggestion.

**Review by mikejolley — APPROVED** [review]
